src/main/java/org/opensearch/security/http/HTTPProxyAuthenticator.java

```diff
@@ -71,17 +71,17 @@ public AuthCredentials extractCredentials(final RestRequest request, ThreadConte
             log.debug("RolesHeader {}, value {}", rolesHeader, rolesHeader == null ? null : request.header(rolesHeader));
         }
 
-        if (!Strings.isNullOrEmpty(userHeader) && !Strings.isNullOrEmpty((String) request.header(userHeader))) {
+        if (!Strings.isNullOrEmpty(userHeader) && !Strings.isNullOrEmpty(request.header(userHeader))) {
```

Comment:
Seems like the null check is to prevent using a null object to find the value in the header collection, maybe it would be clearer to rewrite with optional such as the following, what do you think?
```suggestion
        final Optional<String> userHeaderValue = Optional.ofNullable(userHeader).map(request::header);
        if (userHeaderValue.isPresent()) {
```

Node; I think using optional at the lookup of these headers would be useful too to slim down this logic for the debug statements above too.